src/pocketmine/world/generator/object/Tree.php

```diff
@@ -112,7 +112,9 @@ public function placeObject(ChunkManager $world, int $x, int $y, int $z, Random
 		$this->placeTrunk($x, $y, $z, $random, $this->generateChunkHeight($random), $transaction);
 		$this->placeCanopy($x, $y, $z, $random, $transaction);
 
-		$transaction->apply(); //TODO: handle return value on failure
+		if($transaction->validate()){
+			$transaction->apply(); //TODO: handle return value on failure
+		}
```

Comment:
This boilerplate is now going to be needed everywhere a transaction has been used. For one specific use-case, this is unacceptable.